.devcontainer/devcontainer.json

```diff
@@ -1,20 +1,50 @@
 {
-	"name": "Immich devcontainers",
-	"build": {
-		"dockerfile": "Dockerfile",
-		"args": {
-			"BASEIMAGE": "mcr.microsoft.com/devcontainers/typescript-node:22"
-		}
-	},
+	"name": "Immich",
+  "service": "immich-devcontainer",
+  "dockerComposeFile": [
+    "docker-compose.yml",
+    "../docker/docker-compose.dev.yml"
+  ],
 	"customizations": {
 		"vscode": {
+      "settings": {
+        "dart.flutterSdkPath": "/flutter",
+        "editor.formatOnSave": true,
+        "[javascript][typescript][css]": {
+          "editor.defaultFormatter": "esbenp.prettier-vscode",
+          "editor.tabSize": 2,
+          "editor.formatOnSave": true
+        },
+        "[svelte]": {
+          "editor.defaultFormatter": "svelte.svelte-vscode",
+          "editor.tabSize": 2
+        },
+        "svelte.enable-ts-plugin": true,
+        "eslint.validate": ["javascript", "svelte"],
+        "[dart]": {
+          "editor.formatOnSave": true,
+          "editor.selectionHighlight": false,
+          "editor.suggest.snippetsPreventQuickSuggestions": false,
+          "editor.suggestSelection": "first",
+          "editor.tabCompletion": "onlySnippets",
+          "editor.wordBasedSuggestions": "off",
+          "editor.defaultFormatter": "Dart-Code.dart-code"
+        }
+      },
 			"extensions": [
+        "Dart-Code.dart-code",
+        "Dart-Code.flutter",
+				"dbaeumer.vscode-eslint",
+        "dcmdev.dcm-vscode-extension",
+        "esbenp.prettier-vscode",
 				"svelte.svelte-vscode"
 			]
 		}
 	},
 	"forwardPorts": [],
-	"postCreateCommand": "make install-all",
-	"remoteUser": "node"
+	"initializeCommand": "bash .devcontainer/scripts/initializeCommand.sh",
+	"onCreateCommand": "bash .devcontainer/scripts/onCreateCommand.sh",
+	"overrideCommand": true,
+  "workspaceFolder": "/immich",
```

Comment:
I think for compatibility with github codespaces this needs to be `/workspaces/immich`?

Reply:
Tested it, seems not to be a requirement.